### PR TITLE
sink, config (ticdc): enable transaction split and disable safeMode by default

### LINKED ISSUE #7505: Enable transaction split and disable safeMode of a changefeed in TiCDC by defult.
### Is your feature request related to a problem?

as title.

### Describe the feature you'd like

as title.

### Describe alternatives you've considered

_No response_

### Teachability, Documentation, Adoption, Migration Strategy

_No response_

### PR DESCRIPTION
**Pull request by asddongmen**
### What problem does this PR solve?

Issue Number: close #7505 

#### Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
image

### Release note 

```release-note
Enable transaction split and disable safeMode by default.
```

### CODE REVIEW

File: pkg/applier/redo_test.go [hunk 1]
```diff
@@ -234,7 +234,7 @@ func TestApplyDMLs(t *testing.T) {
 	close(ddlEventCh)
 
 	cfg := &RedoApplierConfig{
-		SinkURI: "mysql://127.0.0.1:4000/?worker-count=1&max-txn-row=1&tidb_placement_mode=ignore",
```
Comment (nongfushanquan) [inline pkg/applier/redo_test.go:237]: should we change the test case since the default mode is false?

**Review by sdojjy — APPROVED** [review]

**Review by nongfushanquan — APPROVED** [review]

**Review by zhaoxinyu — APPROVED** [review]

**Pull request merged**